File: Linked-list/linked-list.go
package main

import "fmt"

type Node struct {
	data int
	next *Node
}

type LinkedList struct {
	head *Node
}

func main() {
	ll := LinkedList{}
	ll.addNode(1)
	ll.addNode(2)
	ll.addNode(3)
	ll.addNode(4)
	ll.addNode(5)
	ll.addNode(6)

	ll.show()
}

func (ll *LinkedList) addNode(data int) {
	newNode := &Node{data: data, next: nil}
	if ll.head == nil {
		ll.head = newNode
	} else {
		current := ll.head
		for current.next != nil {
			current = current.next
		}
		current.next = newNode
	}
}

func (ll *LinkedList) removeFirst() int {
	if ll.head == nil {
		return 0
	}
	deletedNode := ll.head
	ll.head = ll.head.next
	deletedNode.next = nil
	return deletedNode.data
}

func (ll *LinkedList) show() {
	current := ll.head
	for current != nil {
		fmt.Print(current.data, " -> ")
		current = current.next
	}
	fmt.Println("nil")
}

File: Linked-list/linked-list.js
class Node {
  constructor(data){
    this.data = data;
    this.next = null;
  }
}

class LinkedList{
  constructor(){
    this.head = null;
  }

  addNode(data){
    const newNode = new Node(data)
    if(this.head === null){
      this.head = newNode
    } else {
      let current = this.head
      while(current.next !== null){
        current = current.next
      }
      current.next = newNode
    }
  }

  deleteFrist(){
    if(this.head===null) return
    const deletedNode = this.head
    this.head = this.head.next
    deletedNode.next = null
    return deletedNode.data
  }

  show(){
    let current = this.head
    while(current !== null){
      console.log(current.data, '->')
      current = current.next
    }
  }
}

const linkedList = new LinkedList()

linkedList.addNode(1)
console.log(linkedList)

linkedList.addNode(2)
linkedList.addNode(3)
linkedList.addNode(4)
linkedList.addNode(5)
console.log(linkedList)

linkedList.deleteFrist()

linkedList.show()
